common/trinary/flex_trit.c

```diff
@@ -428,10 +428,17 @@ size_t flex_trits_from_bytes(flex_trit_t *to_flex_trits, size_t to_len, const by
   return num_trits;
 }
 
-void flex_trit_print(flex_trit_t const *const flex_trits, size_t trits_len) {
-  size_t tryte_len = trits_len / 3;
-  tryte_t tryte_buff[tryte_len + 1];
-  flex_trits_to_trytes(tryte_buff, tryte_len, flex_trits, trits_len, trits_len);
-  tryte_buff[tryte_len] = '\0';
-  printf("%s", tryte_buff);
+void flex_trit_print(flex_trit_t const *flex_trits, size_t trits_len) {
+#define TRYTE_BUFF_SIZE (243 * NUM_TRITS_PER_FLEX_TRIT)
+  tryte_t tryte_buff[TRYTE_BUFF_SIZE + 1];
+  for(; 0 < trits_len;) {
+    size_t chunk_trits_len = (trits_len < 3 * TRYTE_BUFF_SIZE) ? trits_len : (3 * TRYTE_BUFF_SIZE);
+    size_t chunk_tryte_len = chunk_trits_len / 3;
+    // last incomplete tryte is ignored
+    flex_trits_to_trytes(tryte_buff, chunk_tryte_len, flex_trits, chunk_trits_len, chunk_trits_len);
+    tryte_buff[chunk_tryte_len] = '\0';
+    printf("%s", tryte_buff);
+    trits_len -= chunk_trits_len;
+    flex_trits += chunk_trits_len / NUM_TRITS_PER_FLEX_TRIT;
```

Comment:
it's a print function should not change the data from the caller.

Reply:
This change is only local to the function though

Reply:
ha, right. my bad.